src/test/java/com/kodilla/ecommercee/CartEntityTestSuite.java

```diff
@@ -0,0 +1,138 @@
+        Product product2 = new Product(34L,"Product2", "description2", new BigDecimal(837465),new Group());
+        List<Product> products = new ArrayList<>();
+        products.add(product1);
+        products.add(product2);
+        User user = new User();
+        userRepository.save(user);
+        Cart cart = new Cart(1L, user, products);
+        cartRepository.save(cart);
+
+        //When
+        List<Cart> CartsFromDb = (List<Cart>) cartRepository.findAll();
+        Cart cartFromDb = CartsFromDb.get(0);
+        Product productFromCartFromDb = cartFromDb.getProducts().get(1);
+        String name = productFromCartFromDb.getName();
+
+        //Then
+        assertEquals("Product2", name);
+
+        //CleanUp
+        cartRepository.deleteAll();
+    }
+
+    @Test
+    void addProductToCart() {
+        //Given
+        Product product1 = new Product(33L,"Product1", "description1", new BigDecimal(344),  new Group());
+        Product product2 = new Product(34L,"Product2", "description2", new BigDecimal(837465),new Group());
+        List<Product> products = new ArrayList<>();
+        products.add(product1);
+        User user = new User();
+        userRepository.save(user);
+        Cart cart = new Cart(1L, user, products);
+        cartRepository.save(cart);
+
+        //When
+        List<Cart> CartsFromDb = (List<Cart>) cartRepository.findAll();
+        Cart cartFromDb = CartsFromDb.get(0);
+        cartFromDb.getProducts().add(product2);
+        cartRepository.save(cartFromDb);
+        List<Cart> NewCartsFromDb = (List<Cart>) cartRepository.findAll();
+        Cart newCartFromDb = NewCartsFromDb.get(0);
+        String product2Description = newCartFromDb.getProducts().get(1).getDescription();
+
+        //Then
+        assertEquals("description2", product2Description);
+
+        //CleanUp
+        cartRepository.deleteAll();
+    }
+
+    @Test
+    void deleteProductFromCart() {
+        //Given
+        Product product1 = new Product(33L,"Product1", "description1", new BigDecimal(344),  new Group());
+        Product product2 = new Product(34L,"Product2", "description2", new BigDecimal(837465),new Group());
+        List<Product> products = new ArrayList<>();
+        products.add(product1);
+        products.add(product2);
+        User user = new User();
+        userRepository.save(user);
+        Cart cart = new Cart(1L, user, products);
+        cartRepository.save(cart);
+
+        //When
+        List<Cart> cartsFromDb = (List<Cart>) cartRepository.findAll();
+        Cart cartFromDb = cartsFromDb.get(0);
+        cartFromDb.getProducts().remove(1);
+        cartRepository.save(cartFromDb);
+        List<Cart> newCartsFromDb = (List<Cart>) cartRepository.findAll();
+        Cart newCartFromDb = newCartsFromDb.get(0);
+        int numberOfProducts = newCartFromDb.getProducts().size();
+
+        //Then
+        assertEquals(1, numberOfProducts);
+
+        //CleanUp
+        cartRepository.deleteAll();
```

Comment:
I don't think it is necessary to clean up after each test, we use H2 database so it is doing it without even writing i suppose if you delete all cleanup it should work just well, anyway good job

Reply:
Sometimes is doesn't work and you receive message that some child wasn't been removed